server/mediaserver.go

```diff
@@ -290,8 +290,10 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			return nil
 		}
 		if _, exists := s.rtmpConnections[mid]; exists {
-			glog.Errorf("Manifest manifestID=%v already exists for streamID url=%s", mid, url.String())
-			return nil
+			if extmid != mid {
```

Comment:
So, this is saying if we already have an RTMP conn for the webhook manifestID AND the streamID is different from it then we hit an error condition. Otherwise, if the streamID is the same as the webhook manifestID we do not hit an error condition. This avoids the error condition in the concurrent push test, but I think in practice we could still hit this error condition for concurrent pushes with streamIDs that are different from the webhook manifestID (as would be the case when the playbackId is returned by the webhook).

Reply:
Good catch! I completely missed the case with webhook.

I've added an extra conditional check here https://github.com/livepeer/go-livepeer/pull/1672/commits/5326dbc2564cc73e8eafb45346238ec564b68818 which looks for the `extmid` in the keys of `s.internalManifests` - but it might still be the case that if segments are too close together the older segment hasn't yet updated that mapping - while the newer segment is trying to look for it here.

But imo we can proceed for now and fix it later - as this logic might change again once we improve our extmid/mid split in general in #1673 ?

(Haven't added a test case for concurrent pushes for the webhook-case because it will sometimes fail as noted here)

Reply:
I believe it would still be possible to trigger the error condition here (for an existing RTMP conn) when `extmid == mid`. I don't have a better suggestion at the moment thought, so I'm fine with this code as-is as long as we come back and address it more thoroughly for #1673.